lib/engine/connection.rb

```diff
@@ -21,16 +19,21 @@ def self.connect!(hex)
           hex = path.hex
           connection = connection.branch!(path)
           next if connection.paths.include?(path)
-          # FIXME: for intra-node paths
           next if connection.nodes.include?(path.nodes[0])
+          next if connection.nodes.include?(path.nodes[1])
```

Comment:
you can replace these two lines with 

(connection.nodes & path.nodes).any?

Reply:
OK